injector/src/main/java/edu/ucr/cs/riple/injector/location/LocationToJsonVisitor.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2023 Nima Karimipour
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package edu.ucr.cs.riple.injector.location;
+
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+
+/** A visitor that converts a location to a JSON object. */
+public class LocationToJsonVisitor implements LocationVisitor<JSONObject, Void> {
+
+  /** The Keys used to represent a location in JSON format */
+  public enum KEYS {
+    VARIABLES,
+    METHOD,
+    KIND,
+    CLASS,
+    PATH,
+    INDEX
+  }
+
+  @SuppressWarnings("unchecked")
```

Comment:
Why do we need these unchecked suppressions?

Reply:
This is required for putting object of any type in `JSON`. The `JSONObject` implements raw type of `Map` and every call to `put` is risky in view of javac. The only way to resolve this warning is to change the json dependency to `org.json` from `com.googlecode.json-simple:json-simple` since it is compiled with a very old javac (the latest release is for 2012). We have an open PR that is not landed yet as it has a low priority #123 that replaces the dependency with `org.json` and can get rid of all `@SuppressWarnings("unchecked")`.